### PR TITLE
📚 Categories page & dropdown

### PR DESCRIPTION
**Pull request by mamoore1440**
WIP

### CODE REVIEW

File: src/components/ToolTip.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+<script>
+export default {
+  name: 'tool-tip',
+  props: {
+    text: {
+      type: String,
+      required: true
+    },
+    maxLength: {
```
Comment (cal-overflow) [inline src/components/ToolTip.vue:16]: Need to test this file

File: src/components/previews/Post.vue [hunk 2]
```diff
@@ -41,13 +43,21 @@ export default {
...
     },
-    last: {
+    showMinimalContent: {
       type: Boolean,
-      default: false,
+      default: false
+    },
+    classes: {
+      type: String,
+      default: ''
```
Comment (cal-overflow) [inline src/components/previews/Post.vue:60]: Need to test these changes

File: src/components/ToolTip.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <div :class="`flex flex-col items-center my-2 w-80 text-center transition motion-safe:animate-fade-in ${classes ? classes : ''}`">
+    <div class="w-2 h-2 bg-primary-light dark:bg-primary-dark rounded-full"></div>
+    <p>{{content}}</p>
```
Comment (cal-overflow) [inline src/components/ToolTip.vue:4]: Maybe add some custom tailwind font styling here.

File: src/components/previews/Category.vue [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            View all posts
+          </nuxt-link>
+        </div>
+      </div>
+      <div v-else>
+        <p class="text-md">There are not currently any posts for this category</p>
+      </div>
+  </div>
+  <div v-else class="animate-pulse">
+    TODO: Lazy-loading stuff here
```
Comment (cal-overflow) [inline src/components/previews/Category.vue:54]: Implement this

File: src/components/previews/Category.vue [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        .where({ categories: { $contains: this.category.title }})
+        .sortBy('id', 'desc')
+        .limit(4)
+        .fetch()
+        .catch((err) => {
+          this.$nuxt.error({
+            statusCode: 500,
+            message: `There was an error while trying to fetch posts in ${this.category.title}`,
+            error: err
+          });
```
Comment (cal-overflow) [inline src/components/previews/Category.vue:109]: Should we error here or just treat the category as if it doesn't have any posts?

Maybe we put a small message in the CategoryPreview component that says "error fetching posts" or something.

File: src/components/previews/Post.vue (hunk 2 above)
Comment (cal-overflow) [reply src/components/previews/Post.vue:60]: Done

File: src/tailwind.config.js [hunk 6]
```diff
@@ -4,18 +4,18 @@ module.exports = {
...
+        'main-dark': '#191919',         // Page background color in dark mode
+        'menu-light': '#E0E0E1',        // Background color of menu bar in light mode
+        'menu-dark': '#1F1F1F',         // Background color of menu bar in dark mode
+        'card-light': '#EAEAEB',        // Background color of each card in light mode
+        'card-dark': '#262626',         // Background color of each card in dark mode
+        'primary-light': '#A61E17',     // Primary color in light mode (red)
+        'primary-dark': '#00B4E6',      // Primary color in dark mode (light blue)
+        'extra-gray-light': '#D5D5D7',  // Extra gray color in light mode
+        'extra-gray-dark': '#5A5A5E',   // Extra gray color in dark mode
+        'shadow-dark': '#212121',       // Dark shadow color
```
Comment (cal-overflow) [inline src/tailwind.config.js:18]: Bless up 🙏

File: test/unit/components/preview/Post.spec.js [hunk 7]
```diff
@@ -1,28 +1,39 @@
...
+    });
+
+    afterEach(jest.clearAllMocks);
+
+    it('is a Vue instance', () => {
+      expect(wrapper.vm).toBeTruthy();
+    });
+
+    it('renders the placeholder/lazy-loading content', () => {
+      expect(wrapper.findComponent({ref: 'lazy-load-post-preview'}).exists()).toBeTruthy();
```
Comment (cal-overflow) [inline test/unit/components/preview/Post.spec.js:25]:
```suggestion
      expect(wrapper.findComponent({ ref: 'lazy-load-post-preview' }).exists()).toBeTruthy();
```

**Review by cal-overflow — COMMENTED** [review]
Looking good. A few small suggestions/ideas

File: src/tailwind.config.js (hunk 6 above)
Comment (mamoore1440) [reply src/tailwind.config.js:18]: npnp 😩

File: src/components/ToolTip.vue (hunk 1 above)
Comment (mamoore1440) [reply src/components/ToolTip.vue:16]: done

**Comment by cal-overflow** [comment]
> finish PR and cry

is right. [The tests failed](https://github.com/cal-overflow/portfolio/pull/136#issuecomment-1197394427) 😢

File: test/unit/components/preview/Category.spec.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+    it('contains a PostPreview', () => {
+      const postPreviews = wrapper.findAllComponents(PostPreview);
+      expect(postPreviews).toHaveLength(1);
+    });
+
+    it('does not render the nuxt-content', () => {
+      expect(wrapper.html()).not.toContain('<nuxt-content-stub');
+    });
+  });
```
Comment (cal-overflow) [inline test/unit/components/preview/Category.spec.js:178]: Test that the `lazy-loading` is happening and the PR is 💯

Comment (cal-overflow) [reply test/unit/components/preview/Category.spec.js:178]: Refer to how it's done in the [post-preview](https://github.com/mamoore1440/portfolio-Template/blob/99c60287af6e1c9de1cb8ae1c443438759a1ef77/test/unit/components/preview/Post.spec.js#L24-L26).

**Review by cal-overflow — APPROVED** [review]
Great feature and improvements. Thanks for the help making this 👍

**Pull request merged**